Panic in the FL propagation protocol

Running the test `TestService_DownloadStateRunning` causes a panic in the FL propagation protocol on jenkins, but not locally.

The original error message is `E 19/11/07 12:50:37.810571851: (       v3.(*Overlay).TransmitMsg.func1.1: 174) - Panic in call to protocol <SkipchainPropagateFL>.Dispatch() from service <Skipchain> at address local://127.0.0.1:2010: runtime error: invalid memory address or nil pointer dereference` 

Disabling the recovery code in onet produces:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8b4778]

goroutine 515 [running]:
go.etcd.io/bbolt.(*Bucket).Cursor(...)
	/home/jenkins/go/pkg/mod/go.etcd.io/bbolt@v1.3.2/bucket.go:84
go.etcd.io/bbolt.(*Bucket).Get(0x0, 0xc000364a00, 0x20, 0x20, 0x0, 0x0, 0xc0004d93b0)
	/home/jenkins/go/pkg/mod/go.etcd.io/bbolt@v1.3.2/bucket.go:260 +0x48
go.dedis.ch/cothority/v3/skipchain.(*SkipBlockDB).getFromTx(0xc0001dd200, 0xc0005ec0e0, 0xc000364a00, 0x20, 0x20, 0x0, 0x0, 0xc0001d90d0)
	/home/jenkins/cothority/skipchain/struct.go:1278 +0xc7
go.dedis.ch/cothority/v3/skipchain.(*SkipBlockDB).StoreBlocks.func1(0xc0005ec0e0, 0xe16f70, 0xc0005ec0e0)
	/home/jenkins/cothority/skipchain/struct.go:798 +0x8f5
go.etcd.io/bbolt.(*DB).Update(0xc00029a600, 0xc0004d9780, 0x0, 0x0)
	/home/jenkins/go/pkg/mod/go.etcd.io/bbolt@v1.3.2/db.go:688 +0xb7
go.dedis.ch/cothority/v3/skipchain.(*SkipBlockDB).StoreBlocks(0xc0001dd200, 0xc0004d9840, 0x1, 0x1, 0x0, 0x1, 0x20, 0x20, 0x0)
	/home/jenkins/cothority/skipchain/struct.go:796 +0xfa
go.dedis.ch/cothority/v3/skipchain.(*SkipBlockDB).Store(0xc0001dd200, 0xc0004b4c00, 0x0, 0x0, 0x0)
	/home/jenkins/cothority/skipchain/struct.go:912 +0x8c
go.dedis.ch/cothority/v3/skipchain.(*Service).propagateForwardLinkHandler(0xc0002aa0f0, 0xce9640, 0xc000318160, 0x0, 0x0)
	/home/jenkins/cothority/skipchain/skipchain.go:1588 +0x235
go.dedis.ch/cothority/v3/messaging.(*Propagate).Dispatch(0xc0000ea1e0, 0x0, 0x0)
	/home/jenkins/cothority/messaging/propagate.go:191 +0x189f
go.dedis.ch/onet/v3.(*Overlay).TransmitMsg.func1(0xf311c0, 0xc0000ea1e0, 0xc0000de240, 0xc000282680)
	/home/jenkins/onet/overlay.go:171 +0x5d
created by go.dedis.ch/onet/v3.(*Overlay).TransmitMsg
	/home/jenkins/onet/overlay.go:170 +0x72e
exit status 2
```

--- a/skipchain/struct.go
+++ b/skipchain/struct.go
@@ -1275,7 +1275,11 @@ func (db *SkipBlockDB) storeToTx(tx *bbolt.Tx, sb *SkipBlock) error {
 // An error is thrown if marshalling fails.
 // The caller must ensure that this function is called from within a valid transaction.
 func (db *SkipBlockDB) getFromTx(tx *bbolt.Tx, sbID SkipBlockID) (*SkipBlock, error) {
-	val := tx.Bucket([]byte(db.bucketName)).Get(sbID)
+	b := tx.Bucket([]byte(db.bucketName))
+	if b == nil {
+		return nil, errors.New("bucket does not exist")
+	}
+	val := b.Get(sbID)
 	if val == nil {
 		return nil, nil
 	}

